src/lib/coda_lib/coda_lib.ml

```diff
@@ -36,6 +36,8 @@ module type Network_intf = sig
 
   type state_hash
 
+  type protocol_state_hash
```

Comment:
Why are using this type> It seems like when we instantiate this type, it is `State_hash.t`. Why can't we use `state_hash`?

Reply:
I tried, it doesn't accept a state_hash. I would take a second look at it.